api/notifications_and_subscriptions.rb

```diff
@@ -28,3 +28,26 @@
 delete "#{APIPREFIX}/users/:user_id/subscriptions" do |user_id|
   user.unsubscribe(source).to_hash.to_json
 end
+
+get "#{APIPREFIX}/subscriptions/:thread_id" do |thread_id|
+  page = (params['page'] || DEFAULT_PAGE).to_i
+  per_page = (params['per_page'] || DEFAULT_PER_PAGE).to_i
+
+  # Build a query hash based on the query parameters
+  query = {}
+  query[:subscriber_id] = params[:subscriber_id] if params[:subscriber_id]
+  query[:source_id] = thread_id
+  query[:source_type] = params[:source_type] if params[:source_type]
```

Comment:
I think we only need the `source_id`, no?

Reply:
added subscriber_id and source type to provide filtering of all available attributes. They are not used yet

Reply:
From what I understand, here the source type should always be that of a thread, otherwise this will be a general purpose API for subscriptions, but this seems to be focussed on threads. 
If that is the case, you should only have a one subscription between one use and thread, so specifying a subscriber_id might not make sense.